addons/mail/static/src/core/common/persona_model.js

```diff
@@ -68,6 +70,38 @@ export class Persona extends Record {
     get emailWithoutDomain() {
         return this.email.substring(0, this.email.lastIndexOf("@"));
     }
+
+    get avatarUrl() {
+        const urlParams = {};
+        if (this.write_date) {
+            urlParams.unique = this.write_date;
+        }
+        if (this.type === "partner") {
+            return url("/web/image", {
+                field: "avatar_128",
+                id: this.id,
+                model: "res.partner",
+                ...urlParams,
+            });
+        }
+        if (this.type === "guest") {
+            return url("/web/image", {
+                field: "avatar_128",
+                id: this.id,
+                model: "mail.guest",
+                ...urlParams,
+            });
```

Comment:
need a common method to generate all those URLs in a single unified place

Reply:
I don't get what we could win with this, the only thing a dedicated function would remove is the route param.